adapter/internal/oasparser/envoyconf/response_mapper.go

```diff
@@ -22,34 +22,82 @@ import (
 
 	access_logv3 "github.com/envoyproxy/go-control-plane/envoy/config/accesslog/v3"
 	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoy_config_route_v3 "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
 	hcmv3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	"github.com/wso2/product-microgateway/adapter/config"
 	"github.com/wso2/product-microgateway/adapter/internal/err"
+	"github.com/wso2/product-microgateway/adapter/pkg/soaputils"
 	"google.golang.org/protobuf/types/known/structpb"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 )
 
 func getErrorResponseMappers() []*hcmv3.ResponseMapper {
+	conf, _ := config.ReadConfigs()
+	if conf.Adapter.SoapErrorXMLFormatEnabled {
+		return append(getSoapErrorResponseMappers(), getErrorResponseMappersJSON()...)
+	}
+	return getErrorResponseMappersJSON()
+}
+
+func getErrorResponseMappersJSON() []*hcmv3.ResponseMapper {
+	return []*hcmv3.ResponseMapper{
+		genErrorResponseMapperJSON(err.NotFoundCode, err.NotFoundCode, err.NotFoundMessage, err.NotFoundDescription, "NR"),
```

Comment:
Shall we do an improvement to this code?
First let's maintain the map where the response flag is the key and rest of the properties like specific error code, status code and error message remains in a struct. 
We populate it once and after that you could iterate over the map and call your response mappers.

Reply:
fixed in https://github.com/wso2/product-microgateway/pull/2972/commits/02578c32134f02724d937c2ce88c6b64d8e0c505